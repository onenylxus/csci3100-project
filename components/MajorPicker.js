// Import
import React from 'react';
import { Text, View, Dimensions } from 'react-native';
import { Picker } from '@react-native-community/picker';

// Export major picker
export default function MajorPicker({ callback, value }) {
  const windowWidth = Dimensions.get('window').width;

  function styleByDevice(widthOfDevice, component) {
    if (widthOfDevice < 1100) {
      // Style of Small Screen
      switch (component) {
        case 'width':
          return '100%';

        case 'pickerWidth':
          return '100%';

        default:
          break;
      }
    }

    // Style of Large Screen
    switch (component) {
      case 'width':
        return 550;

      case 'pickerWidth':
        return 250;

      default:
        break;
    }
  }

  return (
    <View>
      {/* Major */}
      <View
        style={{
          marginVertical: 40,
          flexDirection: 'column',
          width: styleByDevice(windowWidth, 'width'),
        }}
      >
        <Text>Major:</Text>
        <View>
          <Picker
            style={{
              width: styleByDevice(windowWidth, 'pickerWidth'),
              alignSelf: 'center',
            }}
            selectedValue={value}
            onValueChange={(itemValue) => callback(itemValue)}
          >
            <Picker.Item label="Select an option" value="" />
            <Picker.Item label="Anthropology" value="ANTHN" />
            <Picker.Item label="Architectural Studies" value="ARCHN" />
            <Picker.Item
              label="Artificial Intelligence: Systems and Technologies"
              value="AISTN"
            />
            <Picker.Item label="Bimodal Bilingual Studies" value="BMBLN" />
            <Picker.Item label="Biochemistry" value="BCHEN" />
            <Picker.Item label="Biology" value="BIOLN" />
            <Picker.Item label="Biomedical Engineering" value="BMEGN" />
            <Picker.Item label="Biomedical Sciences" value="BMSCN" />
            <Picker.Item label="Cell and Molecular Biology" value="CMBIN" />
            <Picker.Item label="Chemistry" value="CHEMN" />
            <Picker.Item label="Chinese Medicine" value="BCMEN" />
            <Picker.Item
              label="Chinese Language and Literature"
              value="CHLLN"
            />
            <Picker.Item label="Chinese Studies" value="CHESB" />
            <Picker.Item label="Community Health Practice" value="CHPRN" />
            <Picker.Item label="Computer Engineering" value="CENGN" />
            <Picker.Item label="Computer Science" value="CSCIN" />
            <Picker.Item label="Cultural Management" value="CUMTN" />
            <Picker.Item label="Cultural Studies" value="Cultural Studies" />
            <Picker.Item
              label="Data Science and Policy Studies"
              value="DSPSN"
            />
            <Picker.Item label="Early Childhood Education" value="CHEDN" />
            <Picker.Item label="Earth System Science" value="ESSCN" />
            <Picker.Item label="Economics" value="ECONN" />
            <Picker.Item label="Electronic Engineering" value="ELEGN" />
            <Picker.Item
              label="Energy and Environmental Engineering"
              value="EEENN"
            />
            <Picker.Item label="English" value="ENGEN" />
            <Picker.Item label="Environmental Science" value="ENSCN" />
            <Picker.Item
              label="Exercise Science and Health Education"
              value="ESHEN"
            />
            <Picker.Item label="Financial Technology" value="FTECN" />
            <Picker.Item label="Fine Arts" value="FAASN" />
            <Picker.Item label="Food and Nutritional Sciences" value="FNSCN" />
            <Picker.Item label="Gender Studies" value="GDRSN" />
            <Picker.Item
              label="Geography and Resource Management"
              value="GRMDN"
            />
            <Picker.Item label="Gerontology" value="BSCGN" />
            <Picker.Item label="Global Business Studies" value="IBBAB" />
            <Picker.Item label="Global Communication" value="GCOMN" />
            <Picker.Item label="Global Studies" value="GLBSN" />
            <Picker.Item
              label="Government and Public Administration"
              value="GPADN"
            />
            <Picker.Item label="History" value="History" />
            <Picker.Item label="Hospitality and Real Estate" value="HTMGB" />
            <Picker.Item label="Information Engineering" value="IERGN" />
            <Picker.Item
              label="Insurance, Financial and Actuarial Analysis"
              value="IFAAB"
            />
            <Picker.Item label="Integrated BBA" value="IBBAB" />
            <Picker.Item label="Japanese Studies" value="JASPN" />
            <Picker.Item label="Journalism and Communication" value="COMMN" />
            <Picker.Item label="Law" value="LAWSN" />
            <Picker.Item label="Liberal Studies" value="LSEDN" />
            <Picker.Item label="Linguistics" value="LINGN" />
            <Picker.Item label="Mathematics" value="MATHN" />
            <Picker.Item
              label="Mathematics and Information Engineering"
              value="MIEGN"
            />
            <Picker.Item
              label="Mathematics and Mathematics Education"
              value="BMEDN"
            />
            <Picker.Item
              label="Mechanical and Automation Engineering"
              value="MAEGN"
            />
            <Picker.Item
              label="Medicine Programme (M.B., Ch.B.)"
              value="MEDUN"
            />
            <Picker.Item label="Molecular Biotechnology" value="MBTEN" />
            <Picker.Item label="Music" value="MUSCN" />
            <Picker.Item label="Natural Sciences" value="NSCIN" />
            <Picker.Item label="Nursing" value="NURSN" />
            <Picker.Item label="Pharmacy" value="PHARN" />
            <Picker.Item label="Philosophy" value="PHILN" />
            <Picker.Item
              label="Physical Education, Exercise Science and Health"
              value="PESHN"
            />
            <Picker.Item label="Physics" value="PHYSN" />
            <Picker.Item label="Professional Accountancy" value="PACCN" />
            <Picker.Item label="Psychology" value="PSYCN" />
            <Picker.Item label="Public Health" value="PHPCN" />
            <Picker.Item label="Quantitative Finance" value="QFINN" />
            <Picker.Item label="Religious Studies" value="RELSN" />
            <Picker.Item label="Risk Management Science" value="RMSCN" />
            <Picker.Item label="Social Work" value="SOWKN" />
            <Picker.Item label="Sociology" value="SOCIN" />
            <Picker.Item label="Statistics" value="STATN" />
            <Picker.Item
              label="Systems Engineering and Engineering Management"
              value="SEEMN"
            />
            <Picker.Item label="Theology" value="THEON" />
            <Picker.Item label="Translation" value="TRANN" />
            <Picker.Item label="Urban Studies" value="URSPN" />
          </Picker>
        </View>
      </View>
    </View>
  );
}
