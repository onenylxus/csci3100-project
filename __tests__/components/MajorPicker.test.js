// Import
import React from 'react';
import { render } from '@testing-library/react-native';

// Import target component
import MajorPicker from '../../components/MajorPicker';

// Run jest tests
describe('MajorPicker', () => {
  let element;

  beforeEach(() => {
    // Render
    element = render(<MajorPicker />);
  });

  afterEach(() => {
    element.unmount();
  });

  it('matches snapshot', () => {
    expect(element.toJSON()).toMatchSnapshot();
  });
});
