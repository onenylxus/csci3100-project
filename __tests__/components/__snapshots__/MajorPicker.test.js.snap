// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MajorPicker matches snapshot 1`] = `
<View>
  <View
    style={
      Object {
        "flexDirection": "column",
        "width": "100%",
      }
    }
  >
    <Text
      style={
        Object {
          "alignSelf": "flex-start",
          "fontFamily": "Roboto",
          "marginBottom": "2%",
          "marginLeft": "10%",
          "marginTop": "1%",
        }
      }
    >
      Major:
    </Text>
    <View
      style={
        Object {
          "marginBottom": 10,
        }
      }
    >
      <View
        style={
          Object {
            "alignSelf": "center",
            "width": "100%",
          }
        }
      >
        <RNCPicker
          items={
            Array [
              Object {
                "label": "Choose a major",
                "textColor": undefined,
                "value": "",
              },
              Object {
                "label": "Anthropology",
                "textColor": undefined,
                "value": "ANTHN",
              },
              Object {
                "label": "Architectural Studies",
                "textColor": undefined,
                "value": "ARCHN",
              },
              Object {
                "label": "Artificial Intelligence: Systems and Technologies",
                "textColor": undefined,
                "value": "AISTN",
              },
              Object {
                "label": "Bimodal Bilingual Studies",
                "textColor": undefined,
                "value": "BMBLN",
              },
              Object {
                "label": "Biochemistry",
                "textColor": undefined,
                "value": "BCHEN",
              },
              Object {
                "label": "Biology",
                "textColor": undefined,
                "value": "BIOLN",
              },
              Object {
                "label": "Biomedical Engineering",
                "textColor": undefined,
                "value": "BMEGN",
              },
              Object {
                "label": "Biomedical Sciences",
                "textColor": undefined,
                "value": "BMSCN",
              },
              Object {
                "label": "Cell and Molecular Biology",
                "textColor": undefined,
                "value": "CMBIN",
              },
              Object {
                "label": "Chemistry",
                "textColor": undefined,
                "value": "CHEMN",
              },
              Object {
                "label": "Chinese Medicine",
                "textColor": undefined,
                "value": "BCMEN",
              },
              Object {
                "label": "Chinese Language and Literature",
                "textColor": undefined,
                "value": "CHLLN",
              },
              Object {
                "label": "Chinese Studies",
                "textColor": undefined,
                "value": "CHESB",
              },
              Object {
                "label": "Community Health Practice",
                "textColor": undefined,
                "value": "CHPRN",
              },
              Object {
                "label": "Computer Engineering",
                "textColor": undefined,
                "value": "CENGN",
              },
              Object {
                "label": "Computer Science",
                "textColor": undefined,
                "value": "CSCIN",
              },
              Object {
                "label": "Cultural Management",
                "textColor": undefined,
                "value": "CUMTN",
              },
              Object {
                "label": "Cultural Studies",
                "textColor": undefined,
                "value": "CULSN",
              },
              Object {
                "label": "Data Science and Policy Studies",
                "textColor": undefined,
                "value": "DSPSN",
              },
              Object {
                "label": "Early Childhood Education",
                "textColor": undefined,
                "value": "CHEDN",
              },
              Object {
                "label": "Earth System Science",
                "textColor": undefined,
                "value": "ESSCN",
              },
              Object {
                "label": "Economics",
                "textColor": undefined,
                "value": "ECONN",
              },
              Object {
                "label": "Electronic Engineering",
                "textColor": undefined,
                "value": "ELEGN",
              },
              Object {
                "label": "Energy and Environmental Engineering",
                "textColor": undefined,
                "value": "EEENN",
              },
              Object {
                "label": "English",
                "textColor": undefined,
                "value": "ENGEN",
              },
              Object {
                "label": "Environmental Science",
                "textColor": undefined,
                "value": "ENSCN",
              },
              Object {
                "label": "Exercise Science and Health Education",
                "textColor": undefined,
                "value": "ESHEN",
              },
              Object {
                "label": "Financial Technology",
                "textColor": undefined,
                "value": "FTECN",
              },
              Object {
                "label": "Fine Arts",
                "textColor": undefined,
                "value": "FAASN",
              },
              Object {
                "label": "Food and Nutritional Sciences",
                "textColor": undefined,
                "value": "FNSCN",
              },
              Object {
                "label": "Gender Studies",
                "textColor": undefined,
                "value": "GDRSN",
              },
              Object {
                "label": "Geography and Resource Management",
                "textColor": undefined,
                "value": "GRMDN",
              },
              Object {
                "label": "Gerontology",
                "textColor": undefined,
                "value": "BSCGN",
              },
              Object {
                "label": "Global Communication",
                "textColor": undefined,
                "value": "GCOMN",
              },
              Object {
                "label": "Global Studies",
                "textColor": undefined,
                "value": "GLBSN",
              },
              Object {
                "label": "Government and Public Administration",
                "textColor": undefined,
                "value": "GPADN",
              },
              Object {
                "label": "History",
                "textColor": undefined,
                "value": "HISTN",
              },
              Object {
                "label": "Hospitality and Real Estate",
                "textColor": undefined,
                "value": "HTMGB",
              },
              Object {
                "label": "Information Engineering",
                "textColor": undefined,
                "value": "IERGN",
              },
              Object {
                "label": "Insurance, Financial and Actuarial Analysis",
                "textColor": undefined,
                "value": "IFAAB",
              },
              Object {
                "label": "Integrated BBA",
                "textColor": undefined,
                "value": "IBBAB",
              },
              Object {
                "label": "Japanese Studies",
                "textColor": undefined,
                "value": "JASPN",
              },
              Object {
                "label": "Journalism and Communication",
                "textColor": undefined,
                "value": "COMMN",
              },
              Object {
                "label": "Law",
                "textColor": undefined,
                "value": "LAWSN",
              },
              Object {
                "label": "Liberal Studies",
                "textColor": undefined,
                "value": "LSEDN",
              },
              Object {
                "label": "Linguistics",
                "textColor": undefined,
                "value": "LINGN",
              },
              Object {
                "label": "Mathematics",
                "textColor": undefined,
                "value": "MATHN",
              },
              Object {
                "label": "Mathematics and Information Engineering",
                "textColor": undefined,
                "value": "MIEGN",
              },
              Object {
                "label": "Mathematics and Mathematics Education",
                "textColor": undefined,
                "value": "BMEDN",
              },
              Object {
                "label": "Mechanical and Automation Engineering",
                "textColor": undefined,
                "value": "MAEGN",
              },
              Object {
                "label": "Medicine",
                "textColor": undefined,
                "value": "MEDUN",
              },
              Object {
                "label": "Molecular Biotechnology",
                "textColor": undefined,
                "value": "MBTEN",
              },
              Object {
                "label": "Music",
                "textColor": undefined,
                "value": "MUSCN",
              },
              Object {
                "label": "Natural Science",
                "textColor": undefined,
                "value": "NSCIN",
              },
              Object {
                "label": "Nursing",
                "textColor": undefined,
                "value": "NURSN",
              },
              Object {
                "label": "Pharmacy",
                "textColor": undefined,
                "value": "PHARN",
              },
              Object {
                "label": "Philosophy",
                "textColor": undefined,
                "value": "PHILN",
              },
              Object {
                "label": "Physical Education, Exercise Science and Health",
                "textColor": undefined,
                "value": "PESHN",
              },
              Object {
                "label": "Physics",
                "textColor": undefined,
                "value": "PHYSN",
              },
              Object {
                "label": "Professional Accountancy",
                "textColor": undefined,
                "value": "PACCN",
              },
              Object {
                "label": "Psychology",
                "textColor": undefined,
                "value": "PHYCN",
              },
              Object {
                "label": "Public Health",
                "textColor": undefined,
                "value": "PHPCN",
              },
              Object {
                "label": "Quantitative Finance",
                "textColor": undefined,
                "value": "QFINN",
              },
              Object {
                "label": "Religious Studies",
                "textColor": undefined,
                "value": "RELSN",
              },
              Object {
                "label": "Risk Management Science",
                "textColor": undefined,
                "value": "RMSCN",
              },
              Object {
                "label": "Social Work",
                "textColor": undefined,
                "value": "SOWKN",
              },
              Object {
                "label": "Sociology",
                "textColor": undefined,
                "value": "SOCIN",
              },
              Object {
                "label": "Statistics",
                "textColor": undefined,
                "value": "STATN",
              },
              Object {
                "label": "Systems Engineering and Engineering Management",
                "textColor": undefined,
                "value": "SEEMN",
              },
              Object {
                "label": "Theology",
                "textColor": undefined,
                "value": "THEON",
              },
              Object {
                "label": "Translation",
                "textColor": undefined,
                "value": "TRANN",
              },
              Object {
                "label": "Urban Studies",
                "textColor": undefined,
                "value": "URSPN",
              },
            ]
          }
          onChange={[Function]}
          selectedIndex={0}
          style={
            Array [
              Object {
                "height": 216,
              },
              undefined,
            ]
          }
        />
      </View>
    </View>
  </View>
</View>
`;
